core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1885,6 +1885,14 @@ private[spark] class DAGScheduler(
               mapOutputTracker.
                 unregisterMergeResult(shuffleId, reduceId, bmAddress, Option(mapIndex))
             }
+          } else {
+            // Unregister the merge result of <shuffleId, reduceId> if
+            // there is a FetchFailed event and is not a
+            // MetaDataFetchException which is signified by bmAddress being null
+            if (bmAddress != null) {
+              mapOutputTracker.
+                unregisterMergeResult(shuffleId, reduceId, bmAddress, None)
+            }
```

Comment:
Can you also check line 2031 where `removeExecutorAndUnregisterOutputs` is called. The `executorId` for ShuffleMergedChunk is `shuffle-push-merger`. Will that cause any issues? We should probably add a UT for this as well with the `unRegisterOutputOnHostOnFetchFailure` enabled.

Reply:
yep, added a condition to check for the special shuffle-push-merger executor, where we only remove the outputs and not the executor itself. 
added a corresponding UT too.